### PR TITLE
feat: added initialisation flag for IntelliJ Deno Config

### LINKED ISSUE #1998: Deno Config Support for IntelliJ IDEA/ Android Studio in supabase init
**Is your feature request related to a problem? Please describe.**
I have to add Deno support in my intelliJ project manually

**Describe the solution you'd like**
I can contribute to adding a function to the cli to generate the config files for intellij ide.

**Describe alternatives you've considered**
Manually generating the deno config for intellij.

I would like to work on this issue.

### PR DESCRIPTION
**Pull request by yashas-hm**
## What kind of change does this PR introduce?
This change include a feature to initialise supabase with deno config for IntelliJ Idea

## What is the current behavior?
Only VSCode settings generation for deno is present. Have to add Deno support in my intelliJ project manually

## What is the new behavior?
Using `--with-intellij-settings` flag with `supabase init` auto generated deno cofig for IntelliJ IDEA.

- Resolves #1998 
- Closes #1998 

## Additional context
This PR resolves @supabase/cli#1998

### CODE REVIEW

File: internal/init/templates/.idea/deno.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="1">
+  <component name="DenoSettings">
+    <option name="useDeno" value="true" />
+  </component>
+</project>
```
Comment (sweatybridge) [inline internal/init/templates/.idea/deno.xml:6]:
```suggestion
</project>

```

File: internal/init/init.go [hunk 2]
```diff
@@ -156,3 +166,22 @@ func writeVscodeConfig(fsys afero.Fs) error {
...
+
+	fmt.Println("Generated IntelliJ settings in " + utils.Bold(denoPath) + ". Please install the Deno plugin!")
+	return nil
+}
+
+func createDenoConfig(path string, template string, fsys afero.Fs) error {
+	if err := afero.WriteFile(fsys, path, []byte(template), 0644); err != nil {
+		return err
+	}
+	return nil
```
Comment (sweatybridge) [inline internal/init/init.go:186]:
```suggestion
	return afero.WriteFile(fsys, path, []byte(template), 0644)
```

There's also utils.WriteFile

File: internal/init/init.go [hunk 3]
```diff
@@ -156,3 +166,22 @@ func writeVscodeConfig(fsys afero.Fs) error {
...
 }
+
+func writeIntelliJConfig(fsys afero.Fs) error {
+	if err := utils.MkdirIfNotExistFS(fsys, intellijDir); err != nil {
+		return err
+	}
+	if err := createDenoConfig(denoPath, intelliJDeno, fsys); err != nil {
+		return err
+	}
+
```
Comment (sweatybridge) [inline internal/init/init.go:177]: We can replace this block with `utils.WriteFile`

**Review by sweatybridge — APPROVED** [review]
LGTM with some minor nitpicks. Appreciate your contribution!

File: internal/init/init.go (hunk 3 above)
Comment (yashas-hm) [reply internal/init/init.go]: Done. good suggestion thanks.

**Pull request merged**